bundles/org.openhab.binding.freeboxos/src/main/resources/OH-INF/thing/channel-types.xml

```diff
@@ -5,11 +5,11 @@
 	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
 
 	<channel-type id="lcd-brightness" advanced="true">
-		<item-type>Number:Dimensionless</item-type>
+		<item-type unitHint="%">Number:Dimensionless</item-type>
 		<label>Screen Brightness</label>
 		<description>Brightness level of the screen in percent</description>
 		<category>DimmableLight</category>
-		<state pattern="%d %unit%" min="0" max="100"/>
+		<state pattern="%d %%" min="0" max="100"/>
```

Comment:
Why this change?

When a new item is created, it should be created with the provided unit from `unitHint`, unless the user decides otherwise, thus should be displayed with % already.

If another unit (e.g. `ONE`) is chosen by the user, this state pattern will override that and still show %.

If no unit is provided at all (e.g. file-based configuration or when removed by the user in the UI), it might make sense to fall back to % as default display unit, but I'm not sure this outweights the benefit of being able to choose another unit than % and having it displayed?

It's not a big deal, but probably this very question will arise in many other PR's, so we might as well try to common ground now. Before `unitHint` was introduced, I agree it was preferable to explicitly provide the display unit as %.

Reply:
Because our recommendation for this channel is to display a percent, whatever the chosen internal unit for the item.

Reply:
It also avoids a wrong display for users using text file and who do not any item unit 
This is especially important for Number:Dimensionless.

Reply:
> It also avoids a wrong display for users using text file and who do not set any unit to the item.

Yes, that's what I meant by:
> > If no unit is provided at all (e.g. file-based configuration or when removed by the user in the UI), it might make sense to fall back to % as default display unit, but I'm not sure this outweights the benefit of being able to choose another unit than % and having it displayed?

Example:
```java
Dimmer ScreenBrightness { channel="freeboxos:...:lcd-brightness", unit="one" }
```

But, of course this can then again be overwritten:
```java
Dimmer ScreenBrightness "Screen Brightness [%d one]" { channel="freeboxos:...:lcd-brightness", unit="one" }
```

It's probably an unlikely scenario anyway, so let's keep it as is.

Reply:
My original opinion was that the second item should not be shown as %, since the user explicitly decided to use unit `one`. But I buy the argument that this is the internal unit, so the user should make a second decision as well regarding the display unit. And to set this to `one` also, the state pattern should be overridden as well.

Conclusion: 👍

Reply:
I just extended my previous message to cover all cases with item unit when using file.

Reply:
And I just reminded myself where my original opinion came from: This logic is different from any other UoM channel types using `%unit%`. As an example, for `Number:Temperature`, channel types most often use `%unit%`. So the display unit is derived from the internal unit set by the user (or defaulted by the regional settings). If binding provides °F and unit is set to °C, the displayed unit is °C as well, without the user having to set both unit and pattern.

Therefore, while I'm fine with the change now, I still think `%unit%` is acceptable as well, and has its own advantages. So when reviewing PR's I think we should accept both, and let the contributors make an informed decision.

Reply:
It is important for Number:Dimensionless because noone expects "one" as displayed unit.
For Number:Temperature, it is better to keep "%unit%".

Reply:
> It is important for Number:Dimensionless because noone expects "one" as displayed unit.

That should be a problem of the past, since we now have the unit hint to take care of this, so the unit should be set correctly when the user creates the item. Thus, `one` is not expected unless the user explicitly sets that (after getting a warning). That was one of of the major reasons for proposing openhab/openhab-core#3854. The only exception is file-based configuration where it's the user's responsibility to manually set the unit.

Reply:
I am going to change transfer-bytes also.

Reply:
Ok, finished.

Reply:
> And I just reminded myself where my original opinion came from: This logic is different from any other UoM channel types using `%unit%`, so it feels inconsistent. As an example, for `Number:Temperature`, channel types most often use `%unit%` in the pattern. So the display unit is derived from the internal unit set by the user (or defaulted by the regional settings). If binding provides °F and unit is set to °C, the displayed unit is °C as well, without the user having to set both unit and pattern.
> 
> Therefore, while I'm fine with the change now, I still think `%unit%` is acceptable as well, and has its own advantages. So when reviewing PR's I think we should accept both, and let the contributors make an informed decision.

I also prefer this approach. It avoid the user to have to overload the displayState

Reply:
Which approach do you prefer?